src/main/java/org/jboss/ejb/client/RequestSendFailedException.java

```diff
@@ -54,7 +56,7 @@ public RequestSendFailedException(final String msg) {
      * @param cause the cause
      */
     public RequestSendFailedException(final Throwable cause) {
-        super(cause);
+        super(throwableToException(cause));
```

Comment:
I would rather that this method return "null" when cause is not an Exception, and then call initCause() just below.

Reply:
Thanks for tip! 👍